plugins/polio/budget/serializers.py

```diff
@@ -129,60 +153,65 @@ class Meta:
-        if invalid_rounds:
-            raise serializers.ValidationError(f"The user does not have the permissions for rounds: {invalid_rounds}.")
+        invalid_round_ids = [round.id for round in rounds if round.id not in valid_rounds_ids]
+        if invalid_round_ids:
+            raise serializers.ValidationError(
+                f"The user does not have the permissions for rounds: {invalid_round_ids}."
+            )
 
         if is_new:
-            already_linked_rounds = [round for round in submitted_rounds if round.budget_process]
-            if already_linked_rounds:
+            already_linked_round_ids = [round.id for round in rounds if round.budget_process]
+            if already_linked_round_ids:
                 raise serializers.ValidationError(
-                    f"A BudgetProcess already exists for rounds: {already_linked_rounds}."
+                    f"A BudgetProcess already exists for rounds: {already_linked_round_ids}."
                 )
 
-        rounds_campaigns = {round.campaign_id for round in submitted_rounds}
+        rounds_campaigns = {round.campaign_id for round in rounds}
         if len(rounds_campaigns) > 1:
-            raise serializers.ValidationError(f"Rounds must be from the same campaign: {submitted_rounds}.")
+            raise serializers.ValidationError("Rounds must be from the same campaign.")
 
         return submitted_rounds
 
-    def create(self, validated_data: dict) -> BudgetProcess:
-        request = self.context["request"]
+    def handle_rounds(self, budget_process, rounds_data):
+        for round_data in rounds_data:
+            round_id = round_data.get("id")
+            if round_id:
+                round_instance = Round.objects.get(id=round_id)
+                round_serializer = BudgetProcessUWriteRoundSerializer(round_instance, data=round_data, partial=True)
+                if round_serializer.is_valid():
+                    round_serializer.save(budget_process=budget_process)
 
-        # Create a new `BudgetProcess`.
+    def create(self, validated_data):
+        request = self.context["request"]
+        rounds_data = validated_data.pop("rounds", [])
         validated_data["created_by"] = request.user
         budget_process = super().create(validated_data)
-
-        # Link rounds.
-        rounds_ids = [round.id for round in self.validated_data["rounds"]]
-        Round.objects.filter(id__in=rounds_ids).update(budget_process=budget_process)
-
+        self.handle_rounds(budget_process, rounds_data)
         return budget_process
 
     def update(self, budget_process: BudgetProcess, validated_data: dict) -> BudgetProcess:
+        rounds_data = validated_data.pop("rounds", [])
+
         budget_process = super().update(budget_process, validated_data)
-        # Unlink old rounds.
-        budget_process.rounds.update(budget_process=None)
-        # Link new rounds.
```

Comment:
These comments added a little bit of explanation: "Unlink old rounds", then "Link new rounds".

Tt would have been nice to leave them ;)